packages/jest-cli/src/cli/runCLI.js

```diff
@@ -91,19 +91,43 @@ module.exports = async (
   };
 
   try {
-    const configs = await Promise.all(
-      roots.map(root => readConfig(argv, root)),
-    );
+    let globalConfig;
+    let hasDeprecationWarnings;
+    let configs = [];
+    let config;
+    if (roots.length === 1) {
```

Comment:
Just to remember, we should rename `roots` to `projects` here as this is what we pass here